File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Service/src/IParticipantService.cs
using Ubb.BikeContest.Model;

namespace Ubb.BikeContest.Service;

public interface IParticipantService : IService<long, Participant>
{
    IEnumerable<Participant> GetParticipantsByTeam(long teamId);

    Participant GetParticipantByData(Participant participant);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/Java/services/src/main/java/com/ubb/service/IParticipantService.java
package com.ubb.service;

import com.ubb.model.Participant;

import java.util.Collection;
import java.util.Optional;

public interface IParticipantService extends IService<Long, Participant> {
    Collection<Participant> getParticipantsByTeam(final Long teamID);

    Optional<Participant> getParticipantByData(final Participant participant);
}
